networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1220 @@
+
+    _Forward_ entropy:
+    .. math:: H_f(G_C) = \frac{1}{|M|} \sum_{v_i \in M} \sum_{v_k \in V \setminus \mu} P(v_i \right v_k) \cdot log k_{out}(v_k)
+    _Backward_ entropy:
+    .. math:: H_b(G_C) = \frac{1}{|\mu|} \sum_{v_i \in \mu} \sum_{v_k \in V_C \setminus M} P(v_i \leftarrow v_k) \cdot log k_{in}(v_k)
+    where :math:`\mu` (M) is the set of minimal (maximal) nodes, and :math:`P(v_i \leftarrow v_k)` indicates
+    the probability of transition between :math:`v_k` and :math:`v_i`.
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    .. [2] "Topological reversibility and causality in feed-forward networks."
+      Corominas-Murtra, Bernat, Carlos Rodríguez-Caso, Joaquín Goñi, and Ricard Solé.
+      New Journal of Physics 12, no. 11 (2010): pg 113051
+
+    .. [3] "Measuring the hierarchy of feedforward networks."
+    Corominas-Murtra, Bernat, Carlos Rodríguez-Caso, Joaquin Goni, and Ricard Solé.
+    Chaos: An Interdisciplinary Journal of Nonlinear Science 21, no. 1 (2011): pg 016108.
+
+    """
+    import numpy as np
+
+    if not nx.is_directed_acyclic_graph(DAG):
+        raise nx.NetworkXError(
+            "G must be a directed acyclic graph for graph entropy evaluation"
+        )
+
+    dag = nx.convert_node_labels_to_integers(DAG)
+    # Could be passed in most contexts to reduce redundant computation
+    L_GC = len(recursive_leaf_removal(DAG))
+
+    if forward_entropy:
+        # as w sklearn: B_prime = normalize(nx.to_numpy_array(dag), axis=1, norm='max')  # Row normalization
+        B_prime = _matrix_normalize(nx.to_numpy_array(dag), row_normalize=True)
+        # +1 as k \in ( 1, L(G_C) )
+        P = sum([np.power(B_prime, k) for k in range(1, L_GC + 1)])
+        # TODO: Not so sure about this unless k coincides with the number of steps already taken (and the sum is odd)
+        # (Presently awaiting response from original authors for clarification)
+    else:
+        # as w sklearn: B = normalize(nx.to_numpy_array(dag), axis=0, norm='max')  # Column normalization
+        B = _matrix_normalize(nx.to_numpy_array(dag), row_normalize=False)
+        P = sum([np.power(B, k) for k in range(1, L_GC + 1)])
+
+    boundary_layer = max_min_layers(dag, max_layer=forward_entropy)
+    non_extremal_nodes = set(dag.nodes() - boundary_layer)
+    entropy = 0
+    for layer_node in boundary_layer:
+        for non_extremal_node in non_extremal_nodes:
+            if forward_entropy:
+                entropy += P[layer_node][non_extremal_node] * np.log(
+                    dag.out_degree(layer_node)
+                )  # nan for 0 outdegree
+            else:
+                entropy += P[non_extremal_node][layer_node] * np.log(
+                    dag.in_degree(layer_node)
+                )
+    entropy /= len(boundary_layer)
```

Comment:
```substitution
    deg = dag.out_degree if forward_entropy else dag.in_degree
    e = sum(P[u][v] * np.log(deg(u) for u in boundary_layer for v in non_extremal_nodes)
    entropy = e / len(boundary_layer)
```

Reply:
That's better! I accounted for the transposed indices with `P = P if forward_entropy else P.T` and implemented your change.